src/control/cmd/dmg/pool.go

```diff
@@ -88,6 +89,36 @@ func (d *PoolDestroyCmd) Execute(args []string) error {
 	return poolDestroy(d.log, d.conns, d.Uuid, d.Force)
 }
 
+// PoolSetPropCmd represents the command to set a property on a pool.
+type PoolSetPropCmd struct {
+	logCmd
+	connectedCmd
+	UUID     string `long:"pool" required:"1" description:"UUID of DAOS pool"`
+	Property string `short:"n" long:"name" required:"1" description:"Name of property to be set"`
+	Value    string `short:"v" long:"value" required:"1" description:"Value of property to be set"`
+}
+
+// Execute is run when PoolSetPropCmd subcommand is activated.
+func (c *PoolSetPropCmd) Execute(_ []string) error {
+	req := client.PoolSetPropReq{
+		UUID:     c.UUID,
+		Property: c.Property,
+	}
+
+	req.SetString(c.Value)
+	if numVal, err := strconv.ParseUint(c.Value, 10, 64); err == nil {
+		req.SetNumber(numVal)
+	}
```

Comment:
Yeah. req.Value is an `interface{}` type. We don't usually use this except in rare and well-defined situations because it can defeat the type safety system if misused. In this case I wanted to approximate a union type (Go doesn't have formal support for union). The alternative is to have two (or maybe three) separate variables -- one for each type plus probably a bool to select between them.